integrations/signup_test.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2017 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package integration
```

Comment:
Maybe we can have an hibrid approach: we'd have an "integration" package that would have the setup and cleanup logic, but the tests itself would be on the router packages.

There's one more thing I'd change: Go routers like Macaron have an `ServeHTTP` method, which can be used to simulate HTTP requests created in tests. So no need to run Gitea in a port and do a real request through the network.

These are ideas, we're totally open to discussion. =)